biostudiesclient/auth.py

```diff
@@ -49,7 +54,13 @@ def login(self):
 
         return auth_response
 
-    def login_payload(self):
+    def __set_credentials(self, username, password):
+        if username:
+            self.username = username
+        if password:
+            self.password = password
+
+    def __login_payload(self):
```

Comment:
I think this will produce unexpected results calling login with and then without parameters.

The general documented idea (as it stands) is that:
- calling login with username and password will login as the specified user.
- calling login without username and password will login as the user in the environment variables.

This code means that once you login with a specified user, from then on that user will be used to login if you call without parameters **instead of the user in the environment variables,** which you can now not login with at all.

This could lead to very hard to track down bugs, because the same execution path can have different results.

Reply:
Steps to reproduce:
username and password in environment variables: **bob**

1. login with no parameters
    - will correctly login as **bob**
2. login with parameters for **eve**
    - will correctly login as **eve**
3. login with no parameters
    - will incorrectly login as **eve** instead of **bob**

Reply:
Cheers. I am going to make a test case from the above and make it pass.